### DISCUSSION TITLE
How to extend the SSCHA min of 2x2x2 SC to for e.g. 4x4x2. SC?

### DISCUSSION THREAD

**Discussion by matukumilli** [SSCHAcode/python-sscha#172] [Q&A] [answered]
Dear SSCHA developers,

How to extend the SSCHA minimisation of 2x2x2 supercell to for e.g. 4x4x2.?

Towards the end of 2x2x2 minimisation, looking at the free energy and FC gradients' fluctuating trends, I have a feeling that supercell size in not sufficient. I hope I can use the 2x2x2 last SSCHA dyn to start with 4x4x2 or 3x3x2.

Please suggest a possible work flow.

Regards,
Prasad Matukumilli

**Comment by mesonepigreco** [comment, marked as answer]
Hi Prasad Matukumilli,
You can interpolate to a bigger q mesh the final SSCHA dynamical matrix to the desired supercell:

```
# [...] Load the final SSCHA dynamical matrix as dyn  

# Interpolate dyn in the new supercell
old_supercell = dyn.GetSupercell()
new_supercell = (4,4,2)
new_dyn = dyn.Interpolate(old_supercell, new_supercell)
new_dyn.save_qe("new_starting_dyn_4x4x2_")
```

Optionally, the Interpolate method supports using an auxiliary harmonic calculation in the desired supercell to improve the interpolation quality.

Remember that the target supercell must comply with the symmetries (so if you have a cubic system, the supercell grid should be cubic)
